markets/perps-market/contracts/modules/AsyncOrderCancelModule.sol

```diff
@@ -0,0 +1,237 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.11 <0.9.0;
+
+import "@synthetixio/core-contracts/contracts/utils/ERC2771Context.sol";
+import {IAsyncOrderCancelModule} from "../interfaces/IAsyncOrderCancelModule.sol";
+import {SafeCastU256, SafeCastI256} from "@synthetixio/core-contracts/contracts/utils/SafeCast.sol";
+import {DecimalMath} from "@synthetixio/core-contracts/contracts/utils/DecimalMath.sol";
+import {IPythVerifier} from "../interfaces/external/IPythVerifier.sol";
+import {PerpsAccount, SNX_USD_MARKET_ID} from "../storage/PerpsAccount.sol";
+import {MathUtil} from "../utils/MathUtil.sol";
+import {PerpsMarket} from "../storage/PerpsMarket.sol";
+import {AsyncOrder} from "../storage/AsyncOrder.sol";
+import {Position} from "../storage/Position.sol";
+import {PerpsMarketConfiguration} from "../storage/PerpsMarketConfiguration.sol";
+import {PerpsPrice} from "../storage/PerpsPrice.sol";
+import {GlobalPerpsMarket} from "../storage/GlobalPerpsMarket.sol";
+import {SettlementStrategy} from "../storage/SettlementStrategy.sol";
+import {PerpsMarketFactory} from "../storage/PerpsMarketFactory.sol";
+import {GlobalPerpsMarketConfiguration} from "../storage/GlobalPerpsMarketConfiguration.sol";
+import {IMarketEvents} from "../interfaces/IMarketEvents.sol";
+
+/**
+ * @title Module for cancelling async orders.
+ * @dev See IAsyncOrderCancelModule.
+ */
+contract AsyncOrderCancelModule is IAsyncOrderCancelModule, IMarketEvents {
+    using DecimalMath for int256;
+    using DecimalMath for uint256;
+    using DecimalMath for int64;
+    using PerpsPrice for PerpsPrice.Data;
+    using PerpsAccount for PerpsAccount.Data;
+    using PerpsMarket for PerpsMarket.Data;
+    using AsyncOrder for AsyncOrder.Data;
+    using SettlementStrategy for SettlementStrategy.Data;
+    using PerpsMarketFactory for PerpsMarketFactory.Data;
+    using GlobalPerpsMarket for GlobalPerpsMarket.Data;
+    using GlobalPerpsMarketConfiguration for GlobalPerpsMarketConfiguration.Data;
+    using Position for Position.Data;
+    using SafeCastU256 for uint256;
+    using SafeCastI256 for int256;
+
+    int256 private constant PRECISION = 18;
+
+    /**
+     * @inheritdoc IAsyncOrderCancelModule
+     */
+    function cancelOrder(uint128 accountId) external view {
```

Comment:
this check can only be done in cancelPythOrder (since we need the pyth price time). And is done here https://github.com/Synthetixio/synthetix-v3/blob/c44a0ff3bdfeac7cfc085413362b4d7ccbfdbcc5/markets/perps-market/contracts/modules/AsyncOrderCancelModule.sol#L209-L210

Reply:
we should be checking that the block timestamp is within the settlement window otherwise the price deviation reward isn't eligible.